src/main/java/io/github/cottonmc/energy/api/EnergyStorage.java

```diff
@@ -1,120 +1,48 @@
 package io.github.cottonmc.energy.api;
 
 import io.github.cottonmc.energy.event.PowerGenEvent;
-import io.github.cottonmc.energy.util.ContainerInteraction;
 import io.github.cottonmc.energy.util.PacketTier;
 import net.fabricmc.fabric.util.HandlerArray;
 import net.minecraft.block.entity.BlockEntity;
-import net.minecraft.util.math.Direction;
 
 import java.util.function.Consumer;
 
 public interface EnergyStorage {
```

Comment:
How does splitting things up help? Every energy-capable BE needs to implement EnergyStorage, since everything needs to have an energy buffer. Everything also needs to be able to insert and extract energy, even if it doesn't do so externally.

Reply:
Does a generator needs to be inserted to? Does a machine need to output energy?

Reply:
That is trie, but I feel like we compartmentalized *too* much. Now there's no in-build methods for internal inserting and extracting internally in the base interface. I kinda wanna make a capability-like interface and have EnergyStorage be a handled internal thing, but I'll write that a little later.

Reply:
The base interface shouldn't have *internal. The implementation objects should have *internal.